src/sinks/vector.rs

```diff
@@ -10,9 +10,16 @@ use bytes::{BufMut, Bytes, BytesMut};
 use futures::{future, Future, Sink};
 use prost::Message;
 use serde::{Deserialize, Serialize};
+use snafu::{ResultExt, Snafu};
 use std::net::{SocketAddr, ToSocketAddrs};
 use tokio::net::TcpStream;
 
+#[derive(Debug, Snafu)]
+enum BuildError {
+    #[snafu(display("Unable to resolve DNS for provided address"))]
+    DNSFailure,
```

Comment:
seems like we can also make this common accross the sinks?

Reply:
Done